mmdet/models/losses/dice_loss.py

```diff
@@ -0,0 +1,55 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from ..builder import LOSSES
+
+
+@LOSSES.register_module()
+class DiceLoss(nn.Module):
+
+    def __init__(self, use_sigmoid=True, loss_weight=1.0):
+        """`Dice Loss <https://arxiv.org/abs/1912.04488>`
+
+        Args:
+            use_sigmoid (bool, optional): Whether to the prediction is
+                used for sigmoid or softmax. Defaults to True.
+            loss_weight (float, optional): Weight of loss. Defaults to 1.0.
+        """
+        super(DiceLoss, self).__init__()
+        self.use_sigmoid = use_sigmoid
+        self.loss_weight = loss_weight
+
+    def forward(self, pred, target):
+        """Forward function.
+
+        Args:
+            pred (torch.Tensor or tuple):
+                The prediction,  if torch.Tensor, shape (n, h, w)
+                if tuple, each param is torch.Tensor with shape (n, w, h)
+            target (torch.Tensor): The learning label of the prediction,
+                shape (n, h, w).
+
+        Returns:
+            torch.Tensor: The calculated loss
+        """
+        assert isinstance(pred, torch.Tensor) or isinstance(pred, tuple)
+        if isinstance(pred, tuple):
+            assert len(pred) == 2
+            if self.use_sigmoid:
+                pred = F.sigmoid(pred[0]) * F.sigmoid(pred[1])
+            else:
+                pred = pred[0] + pred[1]
```

Comment:
This part is not general for DiceLoss but for DecoupledSOLO? Should move this logic to the head.

Reply:
I think  `F.sigmoid(p1) + F.sigmoid(p2)`  !=  `F.sigmoid(p1+p2)`
On my first version, I put these logic in the head,  but I should set `use_sigmoid=False`

In this version, I first assert whether the input is `tuple`. If the input is tuple , I use `pred = F.sigmoid(pred[0]) * F.sigmoid(pred[1])`, else, `pred=F.sigmoid(pred)`.

I think it is necessary to set `use_sigmoid=True`, so is there any way to solve this question?